libraries/hatch-server-user-management/src/UserManagementController.ts

```diff
@@ -14,179 +14,243 @@ import {
+      }
     }
   }
   
-  @route.post('/api/startUserRegistration', StartUserRegistrationRequest.apiMetadata)
+  @route.post(UserServiceClientEndpoints.START_USER_REGISTRATION, StartUserRegistrationRequest.apiMetadata)
   public async startUserRegistration(params: BasicRouteParams) {
     this.logger.debug('Starting user registration...');
     try {
       const {username, password, userAttributes} = params.req.body;
       if (!username || !password) {
         const errMsg = 'Missing required field(s), username and password are required';
         this.logger.debug(errMsg);
-        params.res.status(400).send({
+        params.res.status(HttpStatus.BAD_REQUEST).send({
           error: errMsg,
         });
       } else {
         await this.userService.startUserRegistration(username, password, userAttributes);
         this.logger.debug('User registration started');
-        params.res.sendStatus(200);
+        params.res.sendStatus(HttpStatus.OK);
       }
     } catch (err) {
-      this.logger.error('Error creating user: ', err);
-      params.res.status(500).send({
-        error: err,
-      });
+      const errorMessage = err.code ? err.code + ' - ' + err.message : err;
+      this.logger.error('Error starting user registration: ' + errorMessage);
+      if (err.code === UserManagementErrorCodes.USERNAME_EXISTS) {
+        params.res.status(HttpStatus.PRECONDITION_FAILED).send({
+          error: err.code,
+        });
+      } else {
+        params.res.status(HttpStatus.INTERNAL_SERVER_ERROR).send({
+          error: err.code,
+        });
+      }
     }
   }
   
-  @route.post('/api/resendUserRegistrationCode', ResendUserRegistrationCodeRequest.apiMetadata)
+  @route.post(UserServiceClientEndpoints.RESEND_USER_REGISTRATION, ResendUserRegistrationCodeRequest.apiMetadata)
   public async resendUserRegistrationCode(params: BasicRouteParams) {
     this.logger.debug('Resending user registration code...');
     try {
       const {username} = params.req.body;
       if (!username) {
         const errMsg = 'Missing required field, username is required';
         this.logger.debug(errMsg);
-        params.res.status(400).send({
+        params.res.status(HttpStatus.BAD_REQUEST).send({
           error: errMsg,
         });
       } else {
         await this.userService.resendUserRegistrationCode(username);
         this.logger.debug('User registration code resent');
-        params.res.sendStatus(200);
+        params.res.sendStatus(HttpStatus.OK);
       }
     } catch (err) {
-      this.logger.error('Error resending user sign up email: ', err);
-      params.res.status(500).send({
-        error: err,
-      });
+      const errorMessage = err.code ? err.code + ' - ' + err.message : err;
+      this.logger.error('Error resending user registration: ' + errorMessage);
+      if (err.code === UserManagementErrorCodes.USER_NOT_FOUND) {
+        params.res.status(HttpStatus.UNAUTHORIZED).send({
+          error: 'Unauthorized',
```

Comment:
Should use UserManagementErrorCodes.UNAUTHORIZED here and elsewhere rather than 'Unauthorized' string

Reply:
Per offline discussion, we will leave as is for unauthenticated API calls